### PR TITLE
Converted examples to use Test for execution

### PR DESCRIPTION
**Pull request by brendandixon**
This PR cleans up the examples a bit and provides trivial test files to ease executing them.

Signed-off-by: Brendan Dixon <brendand@microsoft.com>

### CODE REVIEW

**Comment by ahmetalpbalkan** [comment]
Hmm you renamed storage.go to create.go (which is way cryptic IMO) and renamed Storage()? to creaate() ? I think it should be createStorageAccount() .

File: arm/README.md [hunk 1]
```diff
@@ -100,14 +100,13 @@ package main
...
   "fmt"
-  "os"
 
   "github.com/azure/azure-sdk-for-go/examples/helpers"
   "github.com/azure/azure-sdk-for-go/arm/storage"
 )
 
-func main() {
-  name := os.Args[1]
+func check_name(args []string) {
```
Comment (ahmetalpbalkan) [inline arm/README.md]: Golang naming conventions :-)

File: arm/examples/create.go [hunk 2]
```diff
@@ -29,7 +29,7 @@ func withWatcher() autorest.SendDecorator {
 	}
 }
 
-func Storage(args []string) {
+func create_account(args []string) {
```
Comment (ahmetalpbalkan) [inline arm/examples/create.go]: Golang naming conventions. :-)

File: arm/README.md [hunk 3]
```diff
@@ -206,14 +205,14 @@ package main
...
   "fmt"
-  "os"
 
   "github.com/azure/azure-sdk-for-go/examples/helpers"
   "github.com/azure/azure-sdk-for-go/arm/storage"
 )
 
-func main() {
-  name := os.Args[1]
+func create_account(args []string) {
```
Comment (ahmetalpbalkan) [inline arm/README.md]: I'm feeling like this func should take two separate string parameters not an arr?

Comment (brendandixon) [inline arm/README.md]: I could change that.

File: arm/README.md (hunk 1 above)
Comment (brendandixon) [inline arm/README.md]: Oops. :)

**Comment by brendandixon** [comment]
@ahmetalpbalkan Since I addressed your (good) comments. I will merge this change.

File: arm/README.md (hunk 1 above)
Comment (ahmetalpbalkan) [inline arm/README.md]: @brendandixon have you merged without fixing intentionally?

File: arm/README.md (hunk 3 above)
Comment (ahmetalpbalkan) [inline arm/README.md]: @brendandixon this is not fixed either.

**Comment by ahmetalpbalkan** [comment]
@brendandixon "good comments"? :smile:

**Pull request merged**